src/Confix.Tool/src/Confix.Tool/ConfigurationFile/WriteConfigurationFileMiddleware.cs

```diff
@@ -0,0 +1,43 @@
+using Confix.Tool.Commands.Logging;
+using Confix.Tool.Common.Pipelines;
+using Confix.Tool.Schema;
+using Confix.Utilities.Json;
+using Spectre.Console;
+
+namespace Confix.Tool.Middlewares;
+
+public sealed class WriteConfigurationFileMiddleware : IMiddleware
+{
+    /// <inheritdoc />
+    public async Task InvokeAsync(IMiddlewareContext context, MiddlewareDelegate next)
+    {
+        await next(context);
```

Comment:
i would expect the Write…Middleware to be placed in the right spot to first do the persisting and later calling next

Reply:
sample

1. read
2. mutate
3. write
4. validate
4.1 complain that the generated stuff doesn't work out of the box

Reply:
done